### PR TITLE
Task 2 - Serve SPA in AWS

### PR DESCRIPTION
**Pull request by machulan**
[Task 2](https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task2-serve-spa-aws/task.md)

[Lection](https://www.youtube.com/watch?v=8M_JoLh7tA4)

**I had a problem with creating distribution in Cloudfront:**
_com.amazonaws.services.cloudfront.model.AccessDeniedException: 
Your account must be verified before you can add new CloudFront resources. 
To verify your account, please contact AWS Support_
**Support team confirmed that my account is verified and they escalated the problem to CloudFront team.**
**P.S. 28.10.20 01:22 UTC+4 Amazon Support Team informed me that since then I can launch CloudFront distributions.**

**Manual configured s3/cloudfront resources**
* S3 link: http://machula-first-app.s3-website-eu-west-1.amazonaws.com/
* CloudFront link: https://d2kpkxjozx1dr3.cloudfront.net/

**Automated configured s3/cloudfront resources**
* S3 link: http://machula-second-app.s3-website-eu-west-1.amazonaws.com
* CloudFront link: https://d15zr8xzdj76yx.cloudfront.net

Screenshot: useless
Done: 24.10.20
Deadline: 24.10.20
Self-mark: 3/5 (thanks to Amazon Web Services) on 24.10.20 and 5/5 on 28.10.20

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,33 +6,49 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "deploy:s3": "serverless client deploy --no-config-change --no-policy-change --no-cors-change",
```
Comment (olekf) [inline package.json]: duplicates 'client:deploy' script

File: package.json [hunk 2]
```diff
@@ -6,33 +6,49 @@
...
+    "build:deploy": "npm run build && npm run deploy:s3",
```
Comment (olekf) [inline package.json]: duplicates 'client:build:deploy' script

File: serverless.yml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+#org: your-org-name
+
+# You can pin your service to only deploy with a specific Serverless version
+# Check out our docs for more details
+frameworkVersion: '2'
+
+provider:
+  name: aws
+  runtime: nodejs12.x
+
```
Comment (olekf) [inline serverless.yml:27]: no region in configuration, the recommendation was to use eu-west-1 region

File: serverless.yml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+plugins:
+  - serverless-finch
+  - serverless-single-page-app-plugin
+
+custom:
+  client:
+    bucketName: machula-second-app
+    distributionFolder: build
+    indexDocument: index.html
```
Comment (olekf) [inline serverless.yml:36]: useless indexDocument  configuration

**Review by olekf — CHANGES_REQUESTED** [review]

**Comment by olekf** [comment]
Please, follow convention for commit messages: https://docs.rs.school/#/git-convention

**Comment by olekf** [comment]
Please, follow convention for PR description: https://docs.rs.school/#/stage2?id=Требования-к-pull-request-pr

File: package.json (hunk 1 above)
Comment (machulan) [reply package.json]: refactored

File: package.json (hunk 2 above)
Comment (machulan) [reply package.json]: refactored

File: serverless.yml (hunk 3 above)
Comment (machulan) [reply serverless.yml:27]: added

**Comment by machulan** [comment]
I have fixed PR according to RS School convention (except branch name).

**Review by olekf — APPROVED** [review]

**Pull request merged**